src/cryptography/x509/base.py

```diff
@@ -438,10 +453,28 @@ def load_der_x509_crl(
 
 
 class CertificateSigningRequestBuilder(object):
-    def __init__(self, subject_name=None, extensions=[], attributes=[]):
+    def __init__(
+        self,
+        subject_name: typing.Optional[Name] = None,
+        extensions: typing.Optional[
+            typing.Iterable[Extension[ExtensionType]]
+        ] = None,
+        attributes: typing.Optional[
+            typing.Iterable[typing.Tuple[ObjectIdentifier, bytes]]
+        ] = None,
```

Comment:
I simply don't get that.

Passing a list as default value for a function is universally frowned upon. For good reason. I get that the parameters aren't documented, but there's no reason not to document them either. It would certainly make the API easier to use.

If you insist, I will change of course, I can only ask you to reconsider.

Reply:
These were never intended to be a public API (as reflected in our docs), and the goal here is to type what already exists, not to change the API. The builder API with methods that validate values is important to using this correctly. If it'd make this more clear I'm happy to do a seperate PR myself to underscore prefix all the names.

The reason lists are looked down on as a default argument is due to unexpected shared mutation, but these values are never mutated.

Reply:
Okay, you're the boss. *shrug*, I've reverted the changes as you requested.